libs/wingsdk/src/core/app.ts

```diff
@@ -55,24 +40,75 @@ export interface AppProps {
    */
   readonly stateFile?: string;
 
-  /**
-   * A custom factory to resolve polycons.
-   * @default - use the default polycon factory included in the Wing SDK
-   */
-  readonly customFactory?: IPolyconFactory;
-
   /**
    * Absolute paths to plugin javascript files.
    * @default - [] no plugins
    */
   readonly plugins?: string[];
 }
 
+/**
+ * A Wing application.
+ */
+export abstract class App extends Construct {
+  /**
+   * Returns the root app.
+   */
+  public static of(scope: Construct): App {
+    if (scope instanceof App) {
+      return scope;
+    }
+
+    if (!scope.node.scope) {
+      throw new Error("Cannot find root app");
+    }
+
+    return App.of(scope.node.scope);
+  }
+
+  /**
+   * Directory where artifacts are synthesized to.
+   */
+  public abstract readonly outdir: string;
+
+  /**
+   * Synthesize the app into an artifact.
+   */
+  public abstract synth(): string;
+
+  /**
+   * Creates a new object of the given FQN.
+   * @param fqn the fqn of the class to instantiate
+   * @param ctor the constructor of the class to instantiate (undefined for abstract classes)
+   * @param scope the scope of the resource
+   * @param id the id of the resource
+   * @param args the arguments to pass to the resource
+   * @returns the new instance
+   * @throws if the FQN is not supported
+   */
+  public new(
+    fqn: string,
+    ctor: any | undefined,
```

Comment:
JSII might let you get away with using an actual type here, something like:

```suggestion
    ctor: new (scope: Construct, id: string, ...args: any[]) => Construct | undefined,
```

It only really matters for TS, but hey why not. I did not test this and jsii might yell.

Reply:
I'll try it out. Much better of course

Reply:
Sadly JSII didn't like this.